experiments/dispersy/allchannel_module.py

```diff
@@ -36,72 +36,55 @@
 #
 
 # Code:
-from os import path
 from random import choice
 from string import letters
-from sys import path as pythonpath
 from time import time
 
 from twisted.internet.task import LoopingCall
 
-from gumby.experiments.dispersyclient import DispersyExperimentScriptClient, main
-
-
-# TODO(emilon): Fix this crap
-pythonpath.append(path.abspath(path.join(path.dirname(__file__), '..', '..', '..', "./tribler")))
+from gumby.sync import experiment_callback
+from gumby.modules.community_experiment_module import CommunityExperimentModule
 
+from Tribler.community.allchannel.community import AllChannelCommunity
 from Tribler.community.channel.community import ChannelCommunity
 
 
-class AllChannelClient(DispersyExperimentScriptClient):
+class AllChannelModule(CommunityExperimentModule):
 
-    def __init__(self, *argv, **kwargs):
-        from Tribler.community.allchannel.community import AllChannelCommunity
-        super(AllChannelClient, self).__init__(*argv, **kwargs)
-        self.community_class = AllChannelCommunity
+    def __init__(self, experiment):
+        super(AllChannelModule, self).__init__(experiment, AllChannelCommunity)
+        # by default the dispersy providers will not load this one
+        self.session_config.set_enable_channel_search(True)
         self.my_channel = None
         self.joined_community = None
         self.torrentindex = 1
-
         self.join_lc = None
-        self.set_community_kwarg('tribler_session', None)
-
-    def registerCallbacks(self):
-        self.scenario_runner.register(self.create, 'create')
-        self.scenario_runner.register(self.join, 'join')
-        self.scenario_runner.register(self.publish, 'publish')
-        self.scenario_runner.register(self.post, 'post')
-
-    def start_dispersy(self):
-        from Tribler.community.channel.preview import PreviewChannelCommunity
 
-        super(AllChannelClient, self).start_dispersy()
-        self._dispersy.define_auto_load(ChannelCommunity, self._my_member, (), {"tribler_session": None})
-        self._dispersy.define_auto_load(PreviewChannelCommunity, self._my_member, (), {"tribler_session": None})
-
-    def create(self):
+    @experiment_callback
+    def allchannel_create(self):
         self._logger.info("creating-community")
-        self.my_channel = ChannelCommunity.create_community(self._dispersy, self._my_member, tribler_session=None)
+        self.my_channel = ChannelCommunity.create_community(self.dispersy, self.community.my_member,
+                                                            tribler_session=self.session)
```

Comment:
Regarding the AllChannel experiment, we should make sure that we are still using an in-memory database now that we are passing a Tribler session to the `ChannelCommunity` (same holds for `AllChannel`). Having 1000 clients writing everything to db files, will only slow down things.